src/ModelComparison.jl

```diff
@@ -0,0 +1,51 @@
+""" Useful tools for analysing and comparing between multiple PEtriNet models
+"""
+
+module ModelComparison
+
+using AlgebraicPetri
+using Catlab
+using Catlab.CategoricalAlgebra
+import Catlab.CategoricalAlgebra: Subobject
+
+export petri_homomorphisms, compare, PetriSubobject
+
+""" petri_homomorphisms
+Calculates all homomorphisms between two PetriNet models and returns an array
+of ACSetTransformations that describe these homomorphisms.
+
+NOTE:
+This function does restrict to homomorphisms which preserve the transition
+signatures (number of input/output wires).
+"""
+function petri_homomorphisms(p1::T, p2::T; kw...) where {CD, AD, T <: AbstractACSet{CD,AD}}
+  results = ACSetTransformation{CD,AD}[]
+  sigsTS = Set{Vector{Int64}}()
+  homomorphisms(PetriNet(p1), PetriNet(p2); kw...) do transform
+    if all([length(inputs(p1, t)) == length(inputs(p2, transform.components[:T](t))) &&
+            length(outputs(p1, t)) == length(outputs(p2, transform.components[:T](t))) for t in 1:nt(p1)]) &&
+      vcat(transform.components[:T].func, transform.components[:S].func) ∉ sigsTS
```

Comment:
this part is quotienting by the fact that arrows in the petri net are exchangeable, right? If I have multiple parallel arrows between the same state and transition, then the permutations on those arrows are considered equivalences of the petri nets? This really argues further for segregating the transitions by arity/coarity in the schema so that the CSet homs already have to respect these properties.